### ISSUE TITLE
probe httpheaders are duplicated when using istio probe rewriting

### ISSUE THREAD

**Issue by mmerickel** [istio/istio#28466]
**Bug description**

Using a probe (tested with liveness and readiness) with containers running an injected istio-proxy sidecar and probe rewriting enabled causes the `httpHeaders` from the probe to be injected twice into the request.

For example, my test application is receiving the following raw tcp traffic via tcpdump.

I know a tcp echo is not ideal for an HTTP issue, but the problem is with the initial request data so the responses here are irrelevant. I've repro'd it with other apps but this worked as a great minimal example for the bug report.

It's worth noting that if you uncomment the annotation and disable probe rewriting then the request headers are correct, only defined once.

```console
$ tcpdump -A -n -i lo 'tcp port 6543'
... snip ...
GET / HTTP/1.1
Host: localhost:6543
User-Agent: kube-probe/1.18
Accept-Encoding: gzip
Connection: close
Foo: bar
Foo: bar
X-Request-Id: liveness-probe
X-Request-Id: liveness-probe
```

The pod is defined as:

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: echo
  # annotations:
    # sidecar.istio.io/rewriteAppHTTPProbers: "false"
spec:
  containers:
    - name: echo
      image: docker.io/istio/tcp-echo-server:1.2
      imagePullPolicy: IfNotPresent
      args:
        - "6543"
        - "hello"
      ports:
        - containerPort: 6543
      livenessProbe:
        httpGet:
          path: /
          port: 6543
          httpHeaders:
            - name: x-request-id
              value: liveness-probe
            - name: foo
              value: bar
```

[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Expected behavior**

Each header should be defined only once in the probe request - the same way as if istio sidecar injection was disabled (raw kubernetes) or if istio pod rewriting is disabled.

**Steps to reproduce the bug**

- Setup a new kubernetes cluster, install istio.

```console
$ istioctl install --set profile=default
```

- Label the namespace for sidecar injection.

```console
$ kubectl label ns default istio-injection=enabled
```

- Apply the yaml above.

```console
$ kubectl apply -f echo.yaml
```

- Observe traffic into the server via tcpdump, or look at the container logs for the echo server:

```console
$ kubectl logs echo -f -c echo
request: GET / HTTP/1.1
response: hello GET / HTTP/1.1
request: Host: localhost:6543
response: hello Host: localhost:6543
request: User-Agent: kube-probe/1.18
response: hello User-Agent: kube-probe/1.18
request: Accept-Encoding: gzip
response: hello Accept-Encoding: gzip
request: Connection: close
response: hello Connection: close
request: Foo: bar
response: hello Foo: bar
request: Foo: bar
response: hello Foo: bar
request: X-Request-Id: liveness-probe
response: hello X-Request-Id: liveness-probe
request: X-Request-Id: liveness-probe
response: hello X-Request-Id: liveness-probe
request:
response: hello
```

**Version (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version` if you used Helm)**

```console
$ istioctl version --remote
client version: 1.7.4
control plane version: 1.7.4
data plane version: 1.7.4 (2 proxies)
```

```console
$ kubectl version --short
Client Version: v1.19.3
Server Version: v1.18.8
```

**How was Istio installed?**

```console
$ istioctl install --set profile=default
```

**Environment where bug was observed (cloud vendor, OS, etc)**

I have repro'd the issue on both AWS EKS 1.18 as well as docker desktop for mac 2.4.0.0 (48506) running 1.19.3.

**Comment by Monkeyanator** [comment]
The duplication is happening here: https://github.com/istio/istio/blob/master/pilot/cmd/pilot-agent/status/server.go#L447-L461

**Comment by casey-robertson** [comment]
Any chance this is still happening or somehow reintroduced or exposed in 1.7.7 ?   Had a set of apps with issues post 1.6.9 to 1.7.7 upgrade.  They are our only apps that leverage an http health check with user defined headers.  Had to set `sidecar.istio.io/rewriteAppHTTPProbers: "false"` to resolve.  Verified our sidecar injector is set to `true` and has been for some time.

**Comment by Monkeyanator** [comment]
Unfortunately the fix never got backported to `1.7` or `1.8`, I'll get it into the next patch releases for both :+1:

**Comment by casey-robertson** [comment]
> Unfortunately the fix never got backported to `1.7` or `1.8`, I'll get it into the next patch releases for both 👍

Appreciate the quick reply!  What's the best way to tell if a given PR was ever brought into a release?  I was poking around prow.istio.io and searched for the related PR and didn't find it so that gave me an indication.

**Comment by Monkeyanator** [comment]
I was wondering the same thing earlier... I use the source tree and check whether a commit is present in release branch, not sure what alternatives there are to that

**Comment by casey-robertson** [comment]
> I was wondering the same thing earlier... I use the source tree and check whether a commit is present in release branch, not sure what alternatives there are to that
@Monkeyanator 
I see the mention of the merge into the 1.7.8 release.... did this get into 1.8.x ?

**Comment by Monkeyanator** [comment]
> > I see the mention of the merge into the 1.7.8 release.... did this get into 1.8.x ?

[Yes](https://github.com/istio/istio/pull/30955)
